### PR TITLE
Add documentation for subroutes and k8s service visibility tagging

### PR DESCRIPTION
**Pull request by andrew-su**
Add documentation detailing how subroutes are used.
Add documentation detailing how to set visibility of a route via k8s service.

This is documentation relevant to knative starting at v0.8.

I'm looking for guidance on how to make it better if anyone has improvement tips.

### CODE REVIEW

**Comment by samodell** [comment]
Thanks for your PR, @andrew-su ! Before I start my review, can you open this PR against either the current release branch (release-0.9) or the master branch? The master branch would be ideal, but 0.9 works too.

Thanks!

**Comment by andrew-su** [comment]
I've rebased it onto master branch and updated the base.

Looks like the CLA bot is now confused haha

**Comment by samodell** [comment]
Thanks! I can override the bot, so no worries there.

CLA override reason -- bot was happy with this contributor's CLA before the rebase.

**Comment by samodell** [comment]
Hey @tcnghia  -- can you take a look at these changes, or assign someone who can?

**Comment by tcnghia** [comment]
/lgtm
@andrew-su can you please fix the merge conflict?  Thanks

File: docs/serving/cluster-local-route.md [hunk 1]
```diff
@@ -29,8 +29,8 @@ inside the cluster:
 
 To configure a KService to only be available on the cluster-local network (and
 not on the public Internet), you can apply the
-`serving.knative.dev/visibility=cluster-local` label to the KService or Route
-object.
+`serving.knative.dev/visibility=cluster-local` label to the KService, Route or 
+Kubernetes Service object.
```
Comment (tcnghia) [inline docs/serving/cluster-local-route.md]: Kubernetes Service object corresponding to the Route or subroutes?

**Comment by tcnghia** [comment]
/assign @samodell 

for approval

File: docs/serving/cluster-local-route.md (hunk 1 above)
Comment (andrew-su) [reply docs/serving/cluster-local-route.md]: Yep

File: docs/serving/using-subroutes.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Creating and using Subroutes"
+linkTitle: "Configuring cluster-local services"
+weight: 20
+type: "docs"
+---
+
+Subroutes are most effective when used with multiple revisions. When defining a knative service/route, the traffic section of the spec can split between the different revisions. For example:
```
Comment (samodell) [inline docs/serving/using-subroutes.md]: Global: capitalize Knative and Kubernetes.

File: docs/serving/using-subroutes.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- percent: 0
+  revisionName: foo
+- percent: 40
+  revisionName: bar
+- percent: 60
+  revisionName: baz
+```
+
+This allows anyone targeting the main route to have a 0% chance of hitting revision `foo`, 40% chance of hitting revision `bar` and 60% chance of hitting revision `baz`.
+
```
Comment (samodell) [inline docs/serving/using-subroutes.md]:
```suggestion

## Using tags to create target URLs

```

File: docs/serving/using-subroutes.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+As part of the spec there's a key called `tag`. When a `tag` can be applied to the routes, this will generate an address for the specific traffic target.
```
Comment (samodell) [inline docs/serving/using-subroutes.md]:
```suggestion
The spec defines a key called `tag`. When a `tag` is applied to a route, an address for the specific traffic target is created.
```

Comment (samodell) [inline docs/serving/using-subroutes.md]: Is "key" the correct term? Would "attribute" be more accurate? (Not sure, but I don't recall seeing the term key used like this before, so I wanted to ask.)

File: docs/serving/using-subroutes.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+- percent: 0
+  revisionName: foo
+  tag: staging
+- percent: 40
+  revisionName: bar
+- percent: 60
+  revisionName: baz
+```
+
+In the above example, we can access the staging target by accessing `staging-<route name>.<namespace>.<domain>`. Whereas the targets for `bar` and `baz` can only be accessible via the main route `<route name>.<namespace>.<domain>`.
```
Comment (samodell) [inline docs/serving/using-subroutes.md]:
```suggestion
In the above example, you can access the staging target by accessing `staging-<route name>.<namespace>.<domain>`. The targets for `bar` and `baz` can only be accessed using the main route, `<route name>.<namespace>.<domain>`.
```

File: docs/serving/using-subroutes.md (hunk 4 above)
Comment (andrew-su) [reply docs/serving/using-subroutes.md]: "attribute" makes more sense in terms of describing the spec. I'll update it. Thanks!

File: docs/serving/using-subroutes.md (hunk 3 above)
Comment (andrew-su) [reply docs/serving/using-subroutes.md]: This has been added, Github didn't detect it.

**Pull request merged**